source/Fleet.cpp

```diff
@@ -205,19 +206,38 @@ void Fleet::Load(const DataNode &node)
 				variants.clear();
 				total = 0;
 			}
-			variants.emplace_back(child);
-			total += variants.back().weight;
+
+			int weight = 1;
+			string name;
+			bool named = false;
+			if(node.Size() >= 2 + add && !node.IsNumber(1 + add))
+			{
+				name = node.Token(1 + add);
+				named = true;
+			}
+			if(node.Size() >= 2 + add + named)
+				weight = child.Value(1 + add + named);
+
+			if(named)
+				variants.emplace_back(make_pair(GameData::Variants().Get(name), weight));
+			else
+				variants.emplace_back(make_pair(new Variant(child), weight));
```

Comment:
Memory leak - you cannot call `new` without a corresponding `delete`

Reply:
Where would that go, or what would be another way to do this without using `new`?

Reply:
Review the lifetime of the parent container to see the lifetime of a created object. It is undefined behavior or at best implementation defined to refer to/use objects that have gone "out of scope" and that have temporary lifetime - those are the situations when you would need to explicitly manage the object's lifetime via `new`+`delete`.

Most game objects remain in scope (ergo valid) because they are held within some container that is (eventually) a member variable of GameData, so as long as GameData is in scope, they are also in scope.

Reply:
Would it be a bad idea to have `variants` be a `vector<pair<Variant, int>>` instead of having `const Variant *` and dereference the `const Variant *` variants from GameData so that `new/delete` doesn't need to be used here?

Reply:
The better solution is probably to have two different vectors, one with created variants and another with variants pulled from the GameData, as MissionAction does something similar with conversations. (`Conversation conversation` for conversations created in the mission action and `const Conversation *stockConverstion` for conversations created by name and referenced by the MissionAction.)